cypress/e2e/problem_creator.cy.ts

```diff
@@ -9,266 +9,290 @@ describe('Problem creator Test', () => {
     password: 'user',
   };
 
-  beforeEach(() => {
-    cy.clearCookies();
-    cy.clearLocalStorage();
+  // Define test modes
+  const testModes = [
+    {
+      name: 'Authenticated User',
+      isAuthenticated: true,
+      setup: () => {
+        cy.login(loginOptions);
+        profilePage.updatePreferredLanguage('es');
+      },
+      visit: () => {
+        cy.get('[data-nav-problems]').click();
+        cy.get('[data-nav-problems-create-options]').click();
+        cy.get('a[href="/problem/creator/"]').click();
```

Comment:
Okay, in the `problemCreatorPage.ts` file, you already have an intermediate `visit` function that allows you to format the URL you want to visit. So here, you would only need to call it if needed:

```suggestion
        problemCreatorPage.visit();
```